<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>403</width>
    <height>541</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>700</width>
    <height>700</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Калькулятор</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLineEdit" name="displayHistory">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: transparent;</string>
      </property>
      <property name="frame">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLineEdit" name="displayResult">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>20</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: transparent;</string>
      </property>
      <property name="inputMask">
       <string/>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="frame">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="placeholderText">
       <string/>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetFixedSize</enum>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="btnMC">
        <property name="minimumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>MC</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnMR">
        <property name="minimumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>MR</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnMPlus">
        <property name="minimumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>M+</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnMMinus">
        <property name="minimumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>M-</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnMS">
        <property name="minimumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>MS</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnM">
        <property name="minimumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>63</width>
          <height>27</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>M</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <property name="spacing">
       <number>1</number>
      </property>
      <item row="2" column="3">
       <widget class="QPushButton" name="btn9">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>9</string>
        </property>
       </widget>
      </item>
      <item row="5" column="4">
       <widget class="QPushButton" name="btnEqual">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>=</string>
        </property>
       </widget>
      </item>
      <item row="3" column="4">
       <widget class="QPushButton" name="btnMinus">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>−</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="btnSquare">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>x^2</string>
        </property>
       </widget>
      </item>
      <item row="4" column="4">
       <widget class="QPushButton" name="btnPlus">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>+</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="btn4">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>4</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btnPercent">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>%</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QPushButton" name="btn2">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>2</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="btnFraction">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>1/x</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="btn1">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QPushButton" name="btn5">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>5</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="btnRoot">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>sqrt</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QPushButton" name="btn3">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>3</string>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QPushButton" name="btnDot">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>.</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="btn8">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>8</string>
        </property>
       </widget>
      </item>
      <item row="2" column="4">
       <widget class="QPushButton" name="btnMultiply">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>×</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="btn0">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPushButton" name="bntReverse">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>+/-</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="btnC">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>C</string>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QPushButton" name="btn6">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>6</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="btnCE">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>CE</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="btn7">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>7</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="btnBack">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>←</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="btnDivide">
        <property name="minimumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>95</width>
          <height>64</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>÷</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>403</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuThemes">
    <property name="title">
     <string>Themes</string>
    </property>
    <addaction name="actionMacOS_Dark"/>
    <addaction name="actionMacOS_Light"/>
    <addaction name="actionUbuntu_Dark"/>
    <addaction name="actionUbuntu_Light"/>
    <addaction name="actionWindows_11_Dark"/>
    <addaction name="actionWindows_11_Light"/>
   </widget>
   <addaction name="menuThemes"/>
  </widget>
  <action name="actionMacOS_Dark">
   <property name="text">
    <string>MacOS Dark</string>
   </property>
  </action>
  <action name="actionMacOS_Light">
   <property name="text">
    <string>MacOS Light</string>
   </property>
  </action>
  <action name="actionUbuntu_Dark">
   <property name="text">
    <string>Ubuntu Dark</string>
   </property>
  </action>
  <action name="actionUbuntu_Light">
   <property name="text">
    <string>Ubuntu Light</string>
   </property>
  </action>
  <action name="actionWindows_11_Dark">
   <property name="text">
    <string>Windows 11 Dark</string>
   </property>
  </action>
  <action name="actionWindows_11_Light">
   <property name="text">
    <string>Windows 11 Light</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
